### PR TITLE
googleearth: 7.1 -> 7.3 and move to Qt5

### PR DESCRIPTION
**Pull request by peterhoeg**
This is still WIP as it launches with a black screen on a Yoga 2 Pro with intel
graphics and a no-brand desktop using nVidia 650GTX and the nouveau driver but I'd be very interested to see if it runs for other people - especially @chris-martin ref #30491.

Cc: @viric 

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/applications/misc/googleearth/default.nix [hunk 1]
```diff
@@ -20,60 +23,81 @@ let
...
-stdenv.mkDerivation rec {
-  version = "7.1.4.1529";
+
+in stdenv.mkDerivation rec {
   name = "googleearth-${version}";
+  version = "7.3.0.3832";
 
   src = fetchurl {
-    url = "https://dl.google.com/earth/client/current/google-earth-stable_current_${arch}.deb";
+    url = "https://dl.google.com/earth/client/current/google-earth-pro-stable_current_${arch}.deb";
```
Comment (grwlf) [inline pkgs/applications/misc/googleearth/default.nix]: Is it possible to use a mirror here to prevent the silent updates by Google in future?

**Comment by chris-martin** [comment]
Same result for me - The app launches, but there's no globe.
screenshot from 2017-10-18 16-07-39

My machine is an Intel NUC6i7KYK.

**Comment by grwlf** [comment]
Very similar log in my case, and no globe.
```
[grwlf@greyblade ~ ]$ google-earth 

** (googleearth-bin:20832): WARNING **: Error retrieving accessibility bus address: org.freedesktop.DBus.Error.ServiceUnknown: The name org.a11y.Bus was not provided by any .service files
libGL: Can't open configuration file /etc/drirc: No such file or directory.
libGL: Can't open configuration file /home/grwlf/.drirc: No such file or directory.
libGL: pci id for fd 9: 8086:0166, driver i965
libGL: OpenDriver: trying /run/opengl-driver/lib/dri/tls/i965_dri.so
libGL: OpenDriver: trying /run/opengl-driver/lib/dri/i965_dri.so
libGL: Can't open configuration file /etc/drirc: No such file or directory.
libGL: Can't open configuration file /home/grwlf/.drirc: No such file or directory.
libGL: Using DRI3 for screen 0
libGL: Can't open configuration file /etc/drirc: No such file or directory.
libGL: Can't open configuration file /home/grwlf/.drirc: No such file or directory.
```

Also, sometimes it shows a dialog saying that the application wan not installed properly

**Comment by peterhoeg** [comment]
Googling for "google earth black screen linux" unfortunately gives quite a few hits with no definitive solution. Since we cannot build the old version (7.1) as all the download links now point to 7.3 upstream, I'd be inclined to mark this as broken unless somebody has some really bright ideas.

**Comment by grwlf** [comment]
Recently @lebastr published a fix for an older version, which actually attaches *deb package to the ./binaries folder of nixpkgs. 

https://github.com/lebastr/nixpkgs-channels/commit/fcdd9545b2c7cda37b459f80914e7afe328cfb7f

The fix do work, so if one could use third-party hosting to put the *deb and link to it then the problem with the 7.1 version would be solved (not sure about possible license issues).

**Comment by Ma27** [comment]
@peterhoeg I don't know how deep you dug some months ago so I might just repeating things.

According to this thread (https://productforums.google.com/forum/#!topic/maps/IHTPNCzW4EA;context-place=topicsearchin/maps/category$3Alinux%7Csort:relevance%7Cspell:false) Google Earth 7.3 doesn't support several graphic drivers anymore.

We stay at 7.1 (https://drive.google.com/file/d/0Bx0YZ3Xi3rgTcTNjelJwUEVkNDQ/view), however I'm not sure if this is the best solution since we'll have to find a better approach sooner or later (and I'm unfortunately no expert when it comes to graphic drivers :D)

**Comment by TealG** [comment]
Tried this with version updated to 7.3.1, same black screen issue here,
and also managed to crash it by ctrl + click + drag on the black area.
Also tried safe mode, no difference.

output:
```
libGL: Can't open configuration file /etc/drirc: No such file or directory.
libGL: Can't open configuration file /home/map/.drirc: No such file or directory.
libGL: pci id for fd 6: 8086:0166, driver i965
libGL: OpenDriver: trying /run/opengl-driver/lib/dri/tls/i965_dri.so
libGL: OpenDriver: trying /run/opengl-driver/lib/dri/i965_dri.so
libGL: Can't open configuration file /etc/drirc: No such file or directory.
libGL: Can't open configuration file /home/map/.drirc: No such file or directory.
libGL: Using DRI3 for screen 0
libGL: Can't open configuration file /etc/drirc: No such file or directory.
libGL: Can't open configuration file /home/map/.drirc: No such file or directory.

Google Earth has caught signal 11.

We apologize for the inconvenience, but Google Earth has crashed.
 This is a bug in the program, and should never happen under normal
 circumstances. A bug report and debugging data have been written
 to this text file:
```

crashlog:
```
Major Version 7
Minor Version 3
Build Number 0001
Build Date Feb  5 2018
Build Time 23:42:28
OS Type 3
OS Major Version 4
OS Minor Version 9
OS Build Version 78
OS Patch Version 0
Crash Signal 11
Crash Time 1521748128
Up Time 66,5653

Stacktrace from glibc:
./libgoogleearth_pro.so(+0x1bdf3d)[0x7fcc28edcf3d]
./libgoogleearth_pro.so(+0x1f30cc)[0x7fcc28f120cc]
/nix/store/d54amiggq6bw23jw6mdsgamvs6v1g3bh-glibc-2.25-123/lib/libc.so.6(+0x336a0)[0x7fcc289b26a0]
/nix/store/r4sc0dbb8bq2rzdhrz7rzq08hyw3pz1q-googleearth-7.3.1/opt/google/earth/pro/libnavigate.so(_ZN5earth8navigate16ClickToGoToolTip16NearClickedPointERKNS_4evll10MouseEventE+0xa)[0x7fcbf817084a]
/nix/store/r4sc0dbb8bq2rzdhrz7rzq08hyw3pz1q-googleearth-7.3.1/opt/google/earth/pro/libnavigate.so(_ZN5earth8navigate16ClickToGoToolTip9MaybeHideERKNS_4evll10MouseEventE+0x9)[0x7fcbf817df29]
/nix/store/r4sc0dbb8bq2rzdhrz7rzq08hyw3pz1q-googleearth-7.3.1/opt/google/earth/pro/libnavigate.so(_ZN5earth8navigate5state10NavContext11OnMouseMoveERKNS_4evll10MouseEventE+0x1bd)[0x7fcbf81c573d]
/nix/store/r4sc0dbb8bq2rzdhrz7rzq08hyw3pz1q-googleearth-7.3.1/opt/google/earth/pro/libnavigate.so(_ZN5earth8navigate12InputHarness11OnMouseMoveERKNS_4evll10MouseEventE+0x2d)[0x7fcbf816a7fd]
/nix/store/r4sc0dbb8bq2rzdhrz7rzq08hyw3pz1q-googleearth-7.3.1/opt/google/earth/pro/librender.so(_ZN5earth9UIemitterINS_6render14IMouseObserverENS_4evll10MouseEventENS_19EmitterDefaultTraitIS2_S4_EEE14NotifyFilteredEMS2_FvRKS4_ES9_PNS7_10FilterFuncE+0x86)[0x7fcbf850c496]
/nix/store/r4sc0dbb8bq2rzdhrz7rzq08hyw3pz1q-googleearth-7.3.1/opt/google/earth/pro/librender.so(_ZN5earth6render12MouseEmitter6notifyEMNS0_14IMouseObserverEFvRKNS_4evll10MouseEventEES6_+0x5a)[0x7fcbf850543a]
/nix/store/r4sc0dbb8bq2rzdhrz7rzq08hyw3pz1q-googleearth-7.3.1/opt/google/earth/pro/librender.so(_ZN5earth6render12MouseEmitter13SendMouseMoveERKNS_4evll10MouseEventE+0x33)[0x7fcbf8505503]
/nix/store/r4sc0dbb8bq2rzdhrz7rzq08hyw3pz1q-googleearth-7.3.1/opt/google/earth/pro/librender.so(_ZN5earth6render12MouseEmitter9MouseMoveERKNS_4evll10MouseEventE+0x54)[0x7fcbf8505c54]
/nix/store/r4sc0dbb8bq2rzdhrz7rzq08hyw3pz1q-googleearth-7.3.1/opt/google/earth/pro/librender.so(_ZN12RenderWidget14mouseMoveEventEP11QMouseEvent+0x3d)[0x7fcbf850de8d]
/nix/store/ylvsx4c5iwzynhq0zb64qlnpkly2q67d-qtbase-5.9.1/lib/libQt5Widgets.so.5(_ZN7QWidget5eventEP6QEvent+0x210)[0x7fcc279ecb30]
/nix/store/ylvsx4c5iwzynhq0zb64qlnpkly2q67d-qtbase-5.9.1/lib/libQt5Widgets.so.5(_ZN19QApplicationPrivate13notify_helperEP7QObjectP6QEvent+0x9c)[0x7fcc279ab96c]
/nix/store/ylvsx4c5iwzynhq0zb64qlnpkly2q67d-qtbase-5.9.1/lib/libQt5Widgets.so.5(_ZN12QApplication6notifyEP7QObjectP6QEvent+0x10cb)[0x7fcc279b3feb]
/nix/store/ylvsx4c5iwzynhq0zb64qlnpkly2q67d-qtbase-5.9.1/lib/libQt5Core.so.5(_ZN16QCoreApplication15notifyInternal2EP7QObjectP6QEvent+0xaa)[0x7fcc272ba50a]
/nix/store/ylvsx4c5iwzynhq0zb64qlnpkly2q67d-qtbase-5.9.1/lib/libQt5Widgets.so.5(_ZN19QApplicationPrivate14sendMouseEventEP7QWidgetP11QMouseEventS1_S1_PS1_R8QPointerIS0_Eb+0x1dd)[0x7fcc279b26ed]
/nix/store/ylvsx4c5iwzynhq0zb64qlnpkly2q67d-qtbase-5.9.1/lib/libQt5Widgets.so.5(+0x1b5666)[0x7fcc27a06666]
/nix/store/ylvsx4c5iwzynhq0zb64qlnpkly2q67d-qtbase-5.9.1/lib/libQt5Widgets.so.5(+0x1b8043)[0x7fcc27a09043]
/nix/store/ylvsx4c5iwzynhq0zb64qlnpkly2q67d-qtbase-5.9.1/lib/libQt5Widgets.so.5(_ZN19QApplicationPrivate13notify_helperEP7QObjectP6QEvent+0x9c)[0x7fcc279ab96c]
/nix/store/ylvsx4c5iwzynhq0zb64qlnpkly2q67d-qtbase-5.9.1/lib/libQt5Widgets.so.5(_ZN12QApplication6notifyEP7QObjectP6QEvent+0x26b)[0x7fcc279b318b]
/nix/store/ylvsx4c5iwzynhq0zb64qlnpkly2q67d-qtbase-5.9.1/lib/libQt5Core.so.5(_ZN16QCoreApplication15notifyInternal2EP7QObjectP6QEvent+0xaa)[0x7fcc272ba50a]
/nix/store/ylvsx4c5iwzynhq0zb64qlnpkly2q67d-qtbase-5.9.1/lib/libQt5Gui.so.5(_ZN22QGuiApplicationPrivate17processMouseEventEPN29QWindowSystemInterfacePrivate10MouseEventE+0x723)[0x7fcc27fa5873]
/nix/store/ylvsx4c5iwzynhq0zb64qlnpkly2q67d-qtbase-5.9.1/lib/libQt5Gui.so.5(_ZN22QGuiApplicationPrivate24processWindowSystemEventEPN29QWindowSystemInterfacePrivate17WindowSystemEventE+0x135)[0x7fcc27fa7445]
/nix/store/ylvsx4c5iwzynhq0zb64qlnpkly2q67d-qtbase-5.9.1/lib/libQt5Gui.so.5(_ZN22QWindowSystemInterface22sendWindowSystemEventsE6QFlagsIN10QEventLoop17ProcessEventsFlagEE+0xbb)[0x7fcc27f841bb]
/nix/store/ylvsx4c5iwzynhq0zb64qlnpkly2q67d-qtbase-5.9.1/lib/libQt5XcbQpa.so.5(+0x97df0)[0x7fcc0da94df0]
/nix/store/vmyh7mpa2nw3d2fs1lmj523z897gcj6i-glib-2.52.3/lib/libglib-2.0.so.0(g_main_context_dispatch+0x2a7)[0x7fcc1ed7ce37]
/nix/store/vmyh7mpa2nw3d2fs1lmj523z897gcj6i-glib-2.52.3/lib/libglib-2.0.so.0(+0x4a0a0)[0x7fcc1ed7d0a0]
/nix/store/vmyh7mpa2nw3d2fs1lmj523z897gcj6i-glib-2.52.3/lib/libglib-2.0.so.0(g_main_context_iteration+0x2c)[0x7fcc1ed7d14c]
/nix/store/ylvsx4c5iwzynhq0zb64qlnpkly2q67d-qtbase-5.9.1/lib/libQt5Core.so.5(_ZN20QEventDispatcherGlib13processEventsE6QFlagsIN10QEventLoop17ProcessEventsFlagEE+0x5f)[0x7fcc2730a0ff]
/nix/store/ylvsx4c5iwzynhq0zb64qlnpkly2q67d-qtbase-5.9.1/lib/libQt5Core.so.5(_ZN10QEventLoop4execE6QFlagsINS_17ProcessEventsFlagEE+0xfa)[0x7fcc272b924a]
/nix/store/ylvsx4c5iwzynhq0zb64qlnpkly2q67d-qtbase-5.9.1/lib/libQt5Core.so.5(_ZN16QCoreApplication4execEv+0x94)[0x7fcc272c1954]
./libgoogleearth_pro.so(_ZN5earth6client11Application3runEv+0x709)[0x7fcc28f93579]
./libgoogleearth_pro.so(+0x1f32da)[0x7fcc28f122da]
./libgoogleearth_pro.so(earthmain+0x46c)[0x7fcc28f1275c]
/nix/store/d54amiggq6bw23jw6mdsgamvs6v1g3bh-glibc-2.25-123/lib/libc.so.6(__libc_start_main+0xf0)[0x7fcc2899f7e0]
./googleearth-bin[0x4005fe]
```

**Comment by peterhoeg** [comment]
Not worth the hassle.

**Pull request closed without merging**